beacon_node/store/src/iter.rs

```diff
@@ -3,6 +3,25 @@ use std::borrow::Cow;
 use std::sync::Arc;
 use types::{BeaconBlock, BeaconState, BeaconStateError, EthSpec, Hash256, Slot};
 
+/// Implemented for types that have ancestors (e.g., blocks, states) that may be iterated over.
+pub trait AncestorIter<U: Store, I: Iterator> {
+    /// Returns an iterator over the roots of the ancestors of `self`.
+    fn iter_ancestor_roots(&self, store: Arc<U>) -> I;
+}
+
+impl<'a, U: Store, E: EthSpec> AncestorIter<U, BestBlockRootsIterator<'a, E, U>> for BeaconBlock {
+    /// Iterates across all the prior block roots of `self`, starting at the most recent and ending
+    /// at genesis.
+    fn iter_ancestor_roots(&self, store: Arc<U>) -> BestBlockRootsIterator<'a, E, U> {
+        let state = store
+            .get::<BeaconState<E>>(&self.state_root)
+            .expect("state should exist")
+            .expect("store should not error");
```

Comment:
Is this really always guaranteed to succeed? It would require a logic error in how states are stored in order to fail? What if we have a random block from the network that we don't know the state for?

Reply:
Great pickup. I agree this should return an option or result.